### ISSUE TITLE
Compatibility with AngularCompilerPlugin

### ISSUE THREAD

**Issue by neben** [sysgears/webpack-virtual-modules#45]
Is this plugin compatible with `AngularCompilerPlugin`? I keep getting the following error:

```
TypeError: Cannot read property 'data' of undefined
    at setData (/Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack-virtual-modules/index.js:79:15)
    at VirtualFileSystemDecorator.compiler.inputFileSystem._writeVirtualFile (/Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack-virtual-modules/index.js:108:9)
    at VirtualModulesPlugin.writeModule (/Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack-virtual-modules/index.js:59:34)
    at /Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack-virtual-modules/index.js:155:14
    at Array.forEach (<anonymous>)
    at afterResolversHook (/Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack-virtual-modules/index.js:154:40)
    at SyncHook.eval [as call] (eval at create (/Users/ben/projects/sopher.io/sites/webapp/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:5:1)
    at SyncHook.lazyCompileHook (/Users/ben/projects/sopher.io/sites/webapp/node_modules/tapable/lib/Hook.js:154:20)
    at WebpackOptionsApply.process (/Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack/lib/WebpackOptionsApply.js:541:33)
    at webpack (/Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack/lib/webpack.js:57:48)
    at processOptions (/Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack-cli/bin/cli.js:272:16)
    at /Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack-cli/bin/cli.js:364:3
    at Object.parse (/Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack-cli/node_modules/yargs/yargs.js:567:18)
    at /Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack-cli/bin/cli.js:49:8
    at Object.<anonymous> (/Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack-cli/bin/cli.js:366:3)
    at Module._compile (internal/modules/cjs/loader.js:1123:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1143:10)
    at Module.load (internal/modules/cjs/loader.js:972:32)
    at Function.Module._load (internal/modules/cjs/loader.js:872:14)
    at Module.require (internal/modules/cjs/loader.js:1012:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/Users/ben/projects/sopher.io/sites/webapp/node_modules/webpack/bin/webpack.js:156:2)
    at Module._compile (internal/modules/cjs/loader.js:1123:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1143:10)
    at Module.load (internal/modules/cjs/loader.js:972:32)
    at Function.Module._load (internal/modules/cjs/loader.js:872:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47
```

This is an excerpt of the plugins in the webpack.config.js:
```
 plugins: [
    new VirtualModulesPlugin({
      'node_modules/module-foo.js': 'module.exports = { foo: "foo" };'
    }),
    new AngularCompilerPlugin({
      tsConfigPath: path.resolve(TSCONFIG),
      sourceMap: SOURCE_MAPS
    })
]
```

**Comment by larixer** [comment]
@neben I have never checked it with `AngularCompilerPlugin`. I am going to close this issue, since you haven't provided proper reproduction. Feel free to reopen with reproduction repo/steps.

**Comment by ScriptedAlchemy** [comment]
running into the same issue 
https://github.com/manfredsteyer/2020_04_23/tree/static-imports
and chance you can help me out? storage comes across undefined. I also tried it without my abstraction just incase
```js
const AotPlugin = require("@ngtools/webpack").AngularCompilerPlugin;
// TypeScript-Kompilierung + AOT
var VirtualModulesPlugin = require('webpack-virtual-modules');

const ModuleFederationPlugin = require("webpack/lib/container/ModuleFederationPlugin");
const HtmlWebpackPlugin = require("html-webpack-plugin");

const path = require("path");
const webpack = require("webpack");
const VirtualBootstrapPlugin = require("./virtual-bootstrap-plugin");
var virtualModules = new VirtualModulesPlugin({
  'node_modules/module-foo.js': 'module.exports = { foo: "foo" };',
  'node_modules/module-bar.js': 'module.exports = { bar: "bar" };'
});

const shellConfig = {
  entry: "./projects/shell/src/main.ts",
  resolve: {
    mainFields: ["browser", "module", "main"]
  },
  devServer: {
    contentBase: path.join(__dirname, "dist/shell"),
    port: 5000
  },
  module: {
    rules: [
      { test: /\.ts$/, loader: "@ngtools/webpack" },
      { test: /\.html$/, loader: "html-loader" },
      {
        test: /\.css$/,
        use: [
          "style-loader",
          {
            loader: "css-loader",
            options: {
              sourceMap: false,
              import: false
            }
          }
        ]
      },

      {
        test: /\.js$/,
        loader: "@angular-devkit/build-optimizer/webpack-loader",
        options: {
          sourceMap: false
        }
      }
    ]
  },
  plugins: [
virtualModules,
    new ModuleFederationPlugin({
      name: "shell",
      library: { type: "var", name: "shell" },
      remotes: {
        mfe1: "mfe1"
      },
      shared: ["@angular/core", "@angular/common"]
    }),
    new AotPlugin({
      skipCodeGeneration: false,
      tsConfigPath: "./projects/shell/tsconfig.app.json",
      // hostReplacementPaths: {
      //   "./src/environments/environment.ts":
      //   "./src/environments/environment.prod.ts"
      // },
      entryModule: path.resolve(
        __dirname,
        "./projects/shell/src/app/app.module#AppModule"
      )
    }),

    new HtmlWebpackPlugin({
      template: "./projects/shell/src/index.html"
    })
  ],
  output: {
    filename: "[id].[name].js",
    path: __dirname + "/dist/shell",
    chunkFilename: "[id].[chunkhash].js"
  },
  mode: "none"
};

const mfe1Config = {
  entry: ["./projects/mfe1/src/polyfills.ts", "./projects/mfe1/src/main.ts"],
  resolve: {
    mainFields: ["browser", "module", "main"]
  },
  devServer: {
    contentBase: path.join(__dirname, "dist/mfe1"),
    port: 3000
  },
  module: {
    rules: [
      { test: /\.ts$/, loader: "@ngtools/webpack" },
      { test: /\.html$/, loader: "html-loader" },
      {
        test: /\.css$/,
        use: [
          "style-loader",
          {
            loader: "css-loader",
            options: {
              sourceMap: false,
              import: false
            }
          }
        ]
      },

      {
        test: /\.js$/,
        loader: "@angular-devkit/build-optimizer/webpack-loader",
        options: {
          sourceMap: false
        }
      }
    ]
  },
  plugins: [
    new ModuleFederationPlugin({
      name: "mfe1",
      filename: "remoteEntry.js",
      exposes: {
        Component: './projects/mfe1/src/app/app.component.ts'
      },
      library: { type: "var", name: "mfe1" },
      remotes: {
        mfe1: "mfe1"
      },
      shared: ["@angular/core", "@angular/common"]
    }),
    new AotPlugin({
      skipCodeGeneration: false,
      tsConfigPath: "./projects/mfe1/tsconfig.app.json",
      // hostReplacementPaths: {
      //   "./src/environments/environment.ts":
      //     "./src/environments/environment.prod.ts"
      // },
      entryModule: path.resolve(
        __dirname,
        "./projects/mfe1/src/app/app.module#AppModule"
      )
    }),
    new HtmlWebpackPlugin({
      template: "./projects/mfe1/src/index.html"
    })
  ],
  output: {
    publicPath: "http://localhost:3000/",
    filename: "[name].js",
    path: __dirname + "/dist/mfe1",
    chunkFilename: "[id].[chunkhash].js"
  },
  mode: "none"
};

//module.exports = [shellConfig, mfe1Config];
module.exports = shellConfig;
```

**Comment by ScriptedAlchemy** [comment]
exact same issue as mentioned here though.

**Comment by larixer** [comment]
@ScriptedAlchemy Thank you for the reproduction steps! At least now this issue can be regarded as a valid enhancement request.

**Comment by larixer** [comment]
@neben @ScriptedAlchemy The compatibility with AngularCompilerPlugin is added via: #46. Published as `webpack-virtual-modules@0.2.2`

**Comment by edusperoni** [comment]
This seems to be broken again. It outputs an empty script when AngularCompilerPlugin is used.
